integration/deltaspike-data/impl/src/main/java/com/blazebit/persistence/impl/deltaspikedata/impl/handler/EntityViewRepositoryHandler.java

```diff
@@ -0,0 +1,174 @@
+
+        List<V> result = evm.applySetting(
+                createSetting(),
+                cb.where(idAttribute).eq(pk)
+        ).getResultList();
+
+        return result.isEmpty() ? null : result.get(0);
+    }
+
+    @Override
+    public List<V> findAll() {
+        return evm.applySetting(
+                createSetting(),
+                createCriteriaBuilder()
+        ).getResultList();
+    }
+
+    @Override
+    public List<V> findAll(int start, int max) {
+        EntityViewSetting<V, PaginatedCriteriaBuilder<V>> setting = EntityViewSetting.create(viewClass(), start, max);
+        return evm.applySetting(setting, createCriteriaBuilder())
+                .orderByAsc(idAttribute())
+                .getResultList();
+    }
+
+    @Override
+    public List<V> findBy(E e, SingularAttribute<E, ?>... singularAttributes) {
+        return executeExampleQuery(e, 0, 0, false, singularAttributes);
+    }
+
+    @Override
+    public List<V> findBy(E e, int start, int max, SingularAttribute<E, ?>... singularAttributes) {
+        return executeExampleQuery(e, start, max, false, singularAttributes);
+    }
+
+    @Override
+    public List<V> findByLike(E e, SingularAttribute<E, ?>... singularAttributes) {
+        return executeExampleQuery(e, 0, 0, true, singularAttributes);
+    }
+
+    @Override
+    public List<V> findByLike(E e, int start, int max, SingularAttribute<E, ?>... singularAttributes) {
+        return executeExampleQuery(e, start, max, true, singularAttributes);
+    }
+
+    private CriteriaBuilder<E> createCriteriaBuilder() {
+        return cbf.create(context.getEntityManager(), entityClass());
+    }
+
+    private Class<V> viewClass() {
+        return (Class<V>) context.getEntityViewClass();
+    }
+
+    private Class<E> entityClass() {
+        return (Class<E>) context.getEntityClass();
+    }
+
+    private EntityViewSetting<V, CriteriaBuilder<V>> createSetting() {
+        return EntityViewSetting.create((Class<V>) context.getRepositoryMethod().getRepository().getRepositoryEntity().getEntityViewClass());
+    }
+
+    private List<String> extractPropertyNames(SingularAttribute<E, ?>... attributes) {
+        List<String> result = new ArrayList<String>(attributes.length);
+        for (SingularAttribute<E, ?> attribute : attributes) {
+            result.add(attribute.getName());
+        }
+        return result;
+    }
+
+    private List<Property<Object>> extractProperties(SingularAttribute<E, ?>... attributes) {
+        List<String> names = extractPropertyNames(attributes);
+        List<Property<Object>> properties = PropertyQueries.createQuery(context.getEntityClass())
+                .addCriteria(new NamedPropertyCriteria(names.toArray(new String[]{}))).getResultList();
+        return properties;
+    }
+
+    private void prepareWhere(CriteriaBuilder<?> cb, E example, List<Property<Object>> properties, boolean useLikeOperator) {
```

Comment:
Drop the usage of the extract methods and handle the applying to the where builder based on `SingularAttribute` within here. You are converting 3 times right now(SingularAttribute -> String -> Property -> WhereBuilider), just apply that to the builder please. The property type is given by `SingularAttribute.getJavaType` which is probably also much more accurate than what `PropertyQueries` does.

Reply:
The property queries are used to retrieve the parameter values from the example objects.